tests/unit/staff/views/test_users.py

```diff
@@ -588,11 +588,17 @@ def test_userlist_filter_by_invalid_org(rf, staff_area_administrator):
     assert len(response.context_data["object_list"]) == 0
 
 
-def test_userlist_filter_by_role(rf, staff_area_administrator):
-    UserFactory(roles=[OutputPublisher])
-    UserFactory(roles=[ProjectCollaborator])
-
-    request = rf.get("/?role=OutputPublisher")
+@pytest.mark.parametrize(
+    "role",
+    [
+        pytest.param(OutputPublisher, id="global"),
+        pytest.param(ProjectDeveloper, id="local"),
+    ],
+)
+def test_userlist_filter_by_role(rf, staff_area_administrator, role):
+    # first user is staff area administrator; second user is output publisher
```

Comment:
I think this comment is outdated now that the test is parameterised. 
```suggestion
    # Set up two users: one as the staff area administrator to request the view and
    # be filtered out, the other to be included in the response after filtering.
```

Reply:
Good spot!